punishments.js

```diff
@@ -1268,10 +1327,10 @@ Punishments.monitorRoomPunishments = function (user) {
 		if (Config.punishmentautolock && points >= AUTOLOCK_POINT_THRESHOLD) {
 			let rooms = punishments.map(([room]) => room).join(', ');
 			let reason = `Autolocked for having punishments in ${punishments.length} rooms: ${rooms}`;
-			let message = `${user.name} was locked for having punishments in ${punishments.length} rooms: ${punishmentText}`;
+			let message = `${user.name || toId(user)} was locked for having punishments in ${punishments.length} rooms: ${punishmentText}`;
```

Comment:
But ``user.name`` doesn't get cleared until their user object does, and if they got punished, then their user object must not have cleared... (except it's like, a nameblacklist? is that what this is for?)